packages/core/src/api/resolvers/entity/user-entity.resolver.ts

```diff
@@ -32,4 +33,14 @@ export class UserEntityResolver {
             strategy: m instanceof ExternalAuthenticationMethod ? m.strategy : NATIVE_AUTH_STRATEGY_NAME,
         }));
     }
+
+    @ResolveField()
+    async roles(@Ctx() ctx: RequestContext, @Parent() user: User): Promise<Role[]> {
+        let roles: Role[] = [];
+        if (user.roles) {
+            roles = user.roles;
```

Comment:
here you can just have an early return to prevent the rest of the function from executing unnecessary code:

```ts
if (user.roles) {
  return user.roles;
}
```